src/legacy/core_plugins/console/public/src/directives/sense_settings.js

```diff
@@ -30,30 +31,45 @@ require('ui/modules')
       template,
       controllerAs: 'settings',
       controller: function ($scope, $element) {
-        const settings = require('../settings');
+        this.vals = getCurrentSettings();
 
-        this.vals = settings.getCurrentSettings();
-        this.apply = () => {
-          const prevSettings = settings.getAutocomplete();
-          this.vals = settings.updateSettings(this.vals);
-          // Find which, if any, autocomplete settings have changed
+        this.isPollingVisible = () => {
+          const selectedAutoCompleteOptions =
+            Object.keys(this.vals.autocomplete).filter(key => this.vals.autocomplete[key]);
+          return selectedAutoCompleteOptions.length > 0;
+        };
+
+        this.refresh = () => {
+          mappings.retrieveAutoCompleteInfo();
+        };
+
+        this.saveSettings = () => {
+          const prevSettings = getAutocomplete();
+          this.vals = updateSettings(this.vals);
+
+          // Find which, if any, autocomplete settings have changed.
           const settingsDiff = Object.keys(prevSettings).filter(key => prevSettings[key] !== this.vals.autocomplete[key]);
-          if (settingsDiff.length > 0) {
+
+          // Retrieve autocomplete info if the user has changed one of the autocomplete settings or
+          // has turned on polling.
+          if (settingsDiff.length > 0 || getPolling()) {
```

Comment:
I'm not sure the `|| getPolling()` is necessary. I think it's only going to refetch the autocomplete settings if any of them have changed (i.e., if `settingsDiff.length > 0`). So if the user only turns on polling, it will be a no-op.

Reply:
Good catch! I forgot to address this before merging, so I'll fix it in https://github.com/elastic/kibana/pull/39114.